doc/Overview/Mvux/Advanced/Commands.md

```diff
@@ -0,0 +1,209 @@
+
+1. Using code generation
+
+    ### Basic commands
+
+    When creating a method in the Model, it will be generated as an `AsyncCommand` in the Bindable Model.
+
+    For example, when declaring a method of the following signatures:
+
+    ```csharp
+    public void DoWork();
+    public ValueTask DoWork();
+    public ValueTask DoWork(CancellationToken ct);    
+    ```
+
+    A `DoWork` command will be generated in the Bindable Proxy Model. When that command is executed via a button-click etc., the method in the Model will be called:
+
+    ```xml
+    <Button Command="{Binding DoWork}" />
+    ```
+
+    As you may have noticed, the method can be asynchronous and take a `CancellationToken`, but these are not mandatory and commands will be generated for synchronous methods as well.
+
+    ### Adding command parameters
+
+    An additional parameter can be added to the method, which is then assigned with the value of the `CommandParameter` received from the View. For instance, when using a Button and clicking it, the method will be called with the [`Button.CommandParameter`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.primitives.buttonbase.commandparameter) value, given that the type of the value matches the method parameter type. Otherwise the command's `CanExecute` will be false thereby disabling the button:
+
+    ```csharp
+    public void DoWork(double param)
+    {
+        ...
+    }
+    ```
+
+    Then:
+
+    ```xml
+    <Slider x:Name="slider" Minimum="1" Maximum="100"/>
+    <Button Command="{Binding DoWork}" CommandParameter="{Binding Value, ElementName=slider}"/>
+    ```
+
+    Accordingly, if the `CommandParameter` type does not match the parameter type of the method, the button will remain disabled:
+
+    ```xml
+    <Button Command="{Binding DoWork}" CommandParameter="A string, not a double" />
+    ```
+
+    Result:
+
+    ![A disabled refresh button](../Assets/DisabledButton.jpg)
+        
+    ### Command generation rules:
+    
+     - Can be either synchronous or asynchronous (i.e. `public void`, `public async ValueTask`, or `public async Task`)
+     - Any parameter that has a type and name (case-insensitive) matching a Feed or a State in this Model will be evaluated when the Command is invoked and its current latest value will be passed in as an argument.  
+        You'll find this feature to be very powerful in invoking commands by combining data from various Feeds in addition to a command parameter received from the View.
+
+        For example:
+
+        ```csharp
+        public IFeed<int> CounterValue => ...
+
+        public void ResetCounter(int counterValue)
+        {
+            ...
+        }
+        ```
+
+        When the command is executed, because the parameter name `counterValue` matches a feed name in the Model, this parameter will be materialized with the actual value from the Feed when this method is called on command execution.  
```

Comment:
This adds a lot more clarity to the above. 
It looks to be matching a property of the model, rather than "a feed name in the Model".
Does the property have to be a `Feed` or a `State` (or `ISignal`) for this to work?

It also makes me wonder why this is necessary? Why not just reference the `CounterValue` from within the `ResetCounter` method? Why pass it as an argument? Especially as this example suggests that the method will update the property? Or is it treated as a ref parameter and so this is useful for some other reason?